behaviour/graql/language/mutate.feature

```diff
@@ -0,0 +1,698 @@
+    Given graql define
+      """
+      define employment key email;
+      """
+    Given the integrity is validated
+    When get answers of graql query
+      """
+      match $x has email; get;
+      """
+    Then concept identifiers are
+      |     | check | value      |
+      | PER | label | person     |
+      | EMP | label | employment |
+    Then uniquely identify answer concepts
+      | x   |
+      | PER |
+      | EMP |
+
+
+  Scenario: define additional 'key' on a type adds key to it even if it has existing instances
+    Given graql define
+      """
+      define
+      barcode sub attribute, value string;
+      product sub entity, has name;
+      """
+    Given the integrity is validated
+    Given graql insert
+      """
+      insert
+      $x isa product, has name "Cheese";
+      $y isa product, has name "Ham";
+      """
+    Given the integrity is validated
+    When graql define
+      """
+      define
+      product key barcode;
+      """
+    When the integrity is validated
+    When get answers of graql query
+      """
+      match $x has barcode; get;
+      """
+    Then concept identifiers are
+      |     | check | value      |
+      | PRD | label | product    |
+    Then uniquely identify answer concepts
+      | x   |
+      | PRD |
+
+
+  Scenario: define additional 'relates' on a relation type adds roleplayer to it
+    Given graql define
+      """
+      define
+      company sub entity, plays employer;
+      employment relates employer;
+      """
+    Given the integrity is validated
+    When get answers of graql query
+      """
+      match $x relates employer; get;
+      """
+    Then concept identifiers are
+      |     | check | value      |
+      | EMP | label | employment |
+    Then uniquely identify answer concepts
+      | x   |
+      | EMP |
+
+
+  Scenario: define additional 'regex' on attribute type adds regex to it if all existing instances match the regex
+    Given graql insert
+      """
+      insert
+      $x isa person, has name "Alice", has email "alice@grakn.ai";
+      """
+    Given the integrity is validated
+    When graql define
+      """
+      define name regex "^A.*$";
+      """
+    When the integrity is validated
+    When get answers of graql query
+      """
+      match $x regex "^A.*$"; get;
+      """
+    Then concept identifiers are
+      |     | check | value |
+      | NAM | label | name  |
+    Then uniquely identify answer concepts
+      | x   |
+      | NAM |
+
+
+  Scenario: define additional 'regex' on attribute type throws on commit if an existing instance doesn't match the regex
+    Given graql insert
+      """
+      insert
+      $x isa person, has name "Maria", has email "maria@grakn.ai";
+      """
+    Given the integrity is validated
+    Then graql define throws
+      """
+      define name regex "^A.*$";
+      """
+
+
+  Scenario: define additional 'regex' on a long-valued attribute type throws
```

Comment:
The same goes for everything that isn't a string I suppose

Reply:
Yes. But I question whether it's really worth having a test for each individual attribute type. I suppose one could argue that longs and doubles might support regexes in the future?

Nonetheless, not sure how important this is

Reply:
I leave it to you to judge!